Build fails with "An error has occurred in JaCoCo report generation"

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/3912939848/jobs/6688481600#step:11:1404

### Exception stacktrace

`An error has occurred in JaCoCo report generation. Error while creating report: Unknown block type 61.`

```
  [INFO] --- jacoco-maven-plugin:0.8.7:report (post-test) @ pulsar-broker ---
  [INFO] Loading execution data file /home/runner/work/pulsar/pulsar/pulsar-broker/target/jacoco.exec
  [INFO] ------------------------------------------------------------------------
  [INFO] BUILD FAILURE
  [INFO] ------------------------------------------------------------------------
  [INFO] Total time:  14:50 min
  [INFO] Finished at: 2023-01-13T16:38:59Z
  [INFO] ------------------------------------------------------------------------
  Error:  Failed to execute goal org.jacoco:jacoco-maven-plugin:0.8.7:report (post-test) on project pulsar-broker: An error has occurred in JaCoCo report generation. Error while creating report: Unknown block type 61. -> [Help 1]
  Error:  
  Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
  Error:  Re-run Maven using the -X switch to enable full debug logging.
  Error:  
  Error:  For more information about the errors and possible solutions, please read the following articles:
  Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
  Error: Process completed with exit code 1.
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pom.xml
+++ b/pom.xml
@@ -110,6 +110,7 @@ flexible messaging model and an intuitive client API.</description>
     <testRetryCount>1</testRetryCount>
     <testJacocoAgentArgument/>
     <testHeapDumpPath>/tmp</testHeapDumpPath>
+    <surefire.shutdown>kill</surefire.shutdown>
     <docker.organization>apachepulsar</docker.organization>
     <skipSourceReleaseAssembly>false</skipSourceReleaseAssembly>
     <skipBuildDistribution>false</skipBuildDistribution>
@@ -1485,7 +1486,7 @@ flexible messaging model and an intuitive client API.</description>
           </argLine>
           <reuseForks>${testReuseFork}</reuseForks>
           <forkCount>${testForkCount}</forkCount>
-          <shutdown>kill</shutdown>
+          <shutdown>${surefire.shutdown}</shutdown>
           <redirectTestOutputToFile>${redirectTestOutputToFile}</redirectTestOutputToFile>
           <trimStackTrace>false</trimStackTrace>
           <systemPropertyVariables>
@@ -1938,6 +1939,7 @@ flexible messaging model and an intuitive client API.</description>
       <id>coverage</id>
       <properties>
         <testJacocoAgentArgument>@{jacoco.agent.argument}</testJacocoAgentArgument>
+        <surefire.shutdown>exit</surefire.shutdown>
       </properties>
       <build>
         <plugins>

